client/landing/stepper/declarative-flow/internals/hooks/use-sign-up-start-tracking/index.tsx

```diff
@@ -31,20 +32,21 @@ export const useSignUpStartTracking = ( { flow, currentStepRoute }: Props ) => {
 		} ),
 		[ signupStartEventProps, flowVariant ]
 	);
-	const flowName = flow.name;
-	const shouldTrack = flow.isSignupFlow && ( isFirstStep || isSignupStep );
+
 	const removeSignupParam = useCallback( () => {
-		if ( queryParams.has( 'signup' ) ) {
-			queryParams.delete( 'signup' );
+		if ( queryParams.has( 'start' ) ) {
+			queryParams.delete( 'start' );
 			setQuery( queryParams );
 		}
 	}, [ queryParams, setQuery ] );
 
 	useEffect( () => {
-		if ( ! shouldTrack ) {
-			return;
+		if ( shouldTrack ) {
+			recordSignupStart( flowName, ref, extraProps );
+			removeSignupParam();
 		}
-		recordSignupStart( flowName, ref, extraProps || {} );
-		removeSignupParam();
-	}, [ extraProps, flowName, ref, removeSignupParam, shouldTrack ] );
+		// We don't want to re-record the event when the extraProps or query params change.
+		// Please don't add them to the dependency array.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ flowName, ref, removeSignupParam, shouldTrack ] );
```

Comment:
Can we ship this change without introducing the change to the dependency tracking? It feels like that's only going to muddy the waters by making it harder to understand which problem the changes are addressing.